File: tests/fixtures/parsers/UserService.cs
using System.Threading.Tasks;

namespace MyApp.Services
{
    public class UserService
    {
        private readonly IUserRepository _repository;

        public UserService(IUserRepository repository)
        {
            _repository = repository;
        }

        public async Task<User?> GetUserAsync(int id)
        {
            return await _repository.FindByIdAsync(id);
        }

        public List<User> GetAllUsers()
        {
            return _repository.FindAll();
        }
    }
}

File: tests/fixtures/parsers/UserService.java
package com.example.service;

import java.util.List;
import java.util.Optional;

@Service
public class UserService {
    private final UserRepository repository;

    public UserService(UserRepository repository) {
        this.repository = repository;
    }

    public Optional<User> getUser(Long id) {
        return repository.findById(id);
    }

    public List<User> getAllUsers() {
        return repository.findAll();
    }
}
